core/src/main/scala/org/apache/spark/sql/execution/ddl.scala

```diff
@@ -616,3 +616,41 @@ case class UnDeployCommand(alias: String) extends RunnableCommand {
     Nil
   }
 }
+
+case class PutIntoValuesColumnTable(identifier: TableIdentifier,
+    colNames: Option[String],
+    values: Seq[Expression])
+    extends RunnableCommand {
+
+  def convertTypes(value: String, struct: StructField): Any = struct.dataType match {
+    case BinaryType => value.toCharArray().map(ch => ch.toByte)
+    case ByteType => value.toByte
+    case BooleanType => value.toBoolean
+    case DoubleType => value.toDouble
+    case FloatType => value.toFloat
+    case ShortType => value.toShort
+    case DateType => value
+    case IntegerType => value.toInt
+    case LongType => value.toLong
+    case _ => value
+  }
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val snc = new SnappySession(sparkSession.sparkContext)
+    val sc = sparkSession.sparkContext
+    val stats = sparkSession.sharedState
+        .externalCatalog.getTable(snc.getCurrentSchema, identifier.identifier).schema
+    var v1 = values.zipWithIndex.map { case (e, ci) =>
+      val targetType = StringType
+      Cast(e, targetType).eval()
+    }
+    val valuesList = v1.toList
+    val rowRdd = valuesList.zip(stats)
+        .map { case (value, struct) => convertTypes(value.toString, struct) }
```

Comment:
This will fail if not all columns are specified in the put into statement. For example for-
```
create table t1 (col1 int, col2 int, col3 int, col4 int) using column options (key_columns 'col1');
insert into t1 values (1,1,1,1);
put into t1(col1, col2) values (1,2);
```

Reply:
Will modify code for this case